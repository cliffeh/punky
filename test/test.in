1                                      ; 1
1.0				       ; 1.0
"abc"                                  ; "abc"
2                                      ; 2
2.345				       ; 2.345000
"def"                                  ; "def"
3                                      ; 3
3.789e3				       ; 3789.000000
"ghi"                                  ; "ghi"
(+)                                    ; 0
(+ 1)                                  ; 1
(-)                                    ; 0
(- 1)                                  ; -1
(- 7 3)                                ; 4
(*)                                    ; 1
(* 2)                                  ; 2
(* 3 4)                                ; 12
(+ 1 2)                                ; 3
(+ 4 (+ 2 5))                          ; 11
(* 6 (+ 4 3))                          ; 42
(- 10 1 4 3)                           ; 2
(- 10.0 1.0 4.0 3.0)                   ; 2.000000
(/ 9 3)                                ; 3
(/ 64 2 4)                             ; 8
(* 12 (/ 12 3) 2)                      ; 96
(/ (* 4.0 (* 2 3) 2 (+  1 1)) 6)       ; 16.000000
(% 7 3)                                ; 1
(define a 2)                           ; a
a                                      ; 2
(define b 3)                           ; b
b                                      ; 3
(define b 5)                           ; b
b                                      ; 5
(define double (lambda (x) (* x 2)))   ; double
(double 7)                             ; 14
(double (double 4))                    ; 16
(define a (+ a 4))                     ; a
a                                      ; 6
(let ((x 10) (y 20)) (+ y x))          ; 30
(define c 7)                           ; c
(let ((x a) (y b)) (+ x y c))          ; 18
'(a b c d)                             ; (a b c d)
(car '(a b c d))                       ; a
(cdr '(a b c d))                       ; (b c d)
(cons 'a '(b c d))                     ; (a b c d)
(list 'a 'b 'c 'd)                     ; (a b c d)
(if #t 5 7)                            ; 5
(if #f 5 7)                            ; 7
(if #t (+ 7 9))                        ; 16
(if #f (+ 7 9))                        ; ()
(< 5 7)                                ; #t
(> 7 5)                                ; #t
(> 5 7)                                ; #f
(< 7 5)                                ; #f
(< 1 2 3 4)                            ; #t
(> 4 3 2 1)                            ; #t
(= 5 5 5 5)                            ; #t
(>= 5 7)                               ; #f
(<= 5 7)                               ; #t
(<= 7 5)                               ; #f
(>= 7 5)                               ; #t
(append)                               ; ()
(append '() '7)                        ; 7
(append '() '())                       ; ()
(append '(1) '7)                       ; (1 . 7)
(append '(1 2 3) '(4 5 6))             ; (1 2 3 4 5 6)
(append '(1 2 3) '(4 5 6) '7)          ; (1 2 3 4 5 6 . 7)
(append '(1 2 3) '(4 5 6) '(7 8 9))    ; (1 2 3 4 5 6 7 8 9)
(append '(1 2 3) '(4 5 6) '(7 8 9) '0) ; (1 2 3 4 5 6 7 8 9 . 0)
(env)                                  ; ((c . 7) (double . <function>) (b . 5) (a . 6))
(def zzz 27)                           ; zzz
zzz                                    ; 27
