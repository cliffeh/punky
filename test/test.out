1
1.000000
"abc"
2
2.345000
"def"
3
3789.000000
"ghi"
0
1
0
-1
4
1
2
12
3
11
42
2
2.000000
3
8
96
16.000000
a
2
b
3
b
5
double
14
16
30
c
14
(a b c d)
a
(b c d)
(a b c d)
(a b c d)
#t
#f
#t
#f
#f
#f
#f
#t
#f
#t
#t
#t
5
7
16
()
#t
#f
(1 2 3 4 5 6)
"cdefg"
"cde"
0
7
